### PR TITLE
Refactor query/body merging middleware and add test.

### PR DESCRIPTION
**Pull request by rgwozdz**
Refactoring an existing middleware and adding test.

### CODE REVIEW

**Review by sonofflynn89 — APPROVED** [review]

File: lib/middleware/index.js [hunk 1]
```diff
@@ -59,6 +59,19 @@ function tryParse (json) {
   }
 }
 
+function copyBodyToQuery (req, res, next) {
+  const { query = {}, body = {} } = req
+
+  // prefer token to from query to prevent issues with agol proxies
+  const token = query.token || body.token
+
+  req.query = { ...query, ...body, token }
```
Comment (ethanspiro) [inline lib/middleware/index.js:68]: Is there any issue with leaving the token on the body? Im not sure what "prevent issues with agol proxies" really means, but I noticed it was previously deleted.

**Review by ethanspiro — APPROVED** [review]
Looks good, just one question.

File: lib/middleware/index.js (hunk 1 above)
Comment (rgwozdz) [reply lib/middleware/index.js:68]: shoot, I think you're right, I missed it.

Comment (rgwozdz) [reply lib/middleware/index.js:68]: Actually, I'm not so sure.  I think it was just more that body.token was always overriding query.token. @mkdir-ito - can you advise?

**Pull request merged**